### PR TITLE
Add switch for landing gear, pass it to actuators

### PR DESCRIPTION
**Pull request by julianoes**
This adds a switch for the landing gear and adds the landing gear position to `actuator_controls_0[7]`.

Please review @davids5 and @LorenzMeier

### CODE REVIEW

File: src/drivers/boards/tap-v1/board_config.h [hunk 1]
```diff
@@ -195,11 +195,19 @@ __BEGIN_DECLS
 #define GPIO_TIM3_CH4OUT	GPIO_TIM3_CH4OUT_1
 #define DIRECT_PWM_OUTPUT_CHANNELS	4
 
+#define GPIO_TIM3_CH1IN		GPIO_TIM3_CH1IN_1
+#define GPIO_TIM3_CH2IN		GPIO_TIM3_CH2IN_1
+#define GPIO_TIM3_CH3IN		GPIO_TIM3_CH3IN_1
+#define GPIO_TIM3_CH4IN		GPIO_TIM3_CH4IN_1
+#define DIRECT_INPUT_TIMER_CHANNELS  4
+
```
Comment (davids5) [inline src/drivers/boards/tap-v1/board_config.h]: Why were these added?

I do not think these needed. HW wise they do not make sense. Do I have an underlying dependency I need to fix?

File: src/drivers/boards/tap-v1/tap_timer_config.c [hunk 2]
```diff
@@ -58,40 +58,16 @@ __EXPORT const io_timers_t io_timers[MAX_IO_TIMERS] = {
...
-		.gpio_in = 0,
-		.timer_index = 0,
-		.timer_channel = 3,
-		.ccr_offset = STM32_GTIM_CCR3_OFFSET,
-		.masks  = GTIM_SR_CC3IF | GTIM_SR_CC3OF
-	},
-	{
 		.gpio_out = GPIO_TIM3_CH4OUT,
-		.gpio_in = 0,
+		.gpio_in = GPIO_TIM3_CH4IN,
```
Comment (davids5) [inline src/drivers/boards/tap-v1/tap_timer_config.c]: @julianoes  this looks odd - I would expect the .gpio_in to be left as = 0

File: src/drivers/stm32/drv_led_pwm.cpp [hunk 3]
```diff
@@ -85,8 +85,9 @@
 #define rDMAR(_tmr)     REG(_tmr, STM32_GTIM_DMAR_OFFSET)
 #define rBDTR(_tmr)     REG(_tmr, STM32_ATIM_BDTR_OFFSET)
 
-static void             led_pwm_timer_init(unsigned timer);
-static void             led_pwm_timer_set_rate(unsigned timer, unsigned rate);
+
+extern int             io_timer_init_timer(unsigned timer);
+
```
Comment (davids5) [inline src/drivers/stm32/drv_led_pwm.cpp]: When I refactored the original pwm servo code at add input capture. I add the notion of io_timers that the input capture and new pwm servo are then layered on

The drv_led_pwm.cpp looks like a quick solution, by way of a copy of the pwm servo code, done by after the fact. 

Here are the 2 64$ questions: 1) Can the led_pwm code be refactored to do the RGB on top of the pwm servo or io_timer? 2) Would this and further refactoring create unwanted coupling?

File: src/drivers/boards/tap-v1/tap_timer_config.c (hunk 2 above)
Comment (julianoes) [inline src/drivers/boards/tap-v1/tap_timer_config.c]: This was needed for the check here:
https://github.com/PX4/Firmware/blob/78b0d1a01f2f3ebfd33d005d9da5908c72e4843b/src/drivers/stm32/drv_io_timer.c#L290

File: src/drivers/px4fmu/fmu.cpp [hunk 4]
```diff
@@ -3092,7 +3126,14 @@ fmu_main(int argc, char *argv[])
...
 		new_mode = PORT_FULL_PWM;
 
-#if defined(BOARD_HAS_PWM) && BOARD_HAS_PWM >= 6
+#if defined(BOARD_HAS_PWM)
+
+	} else if (!strcmp(verb, "mode_pwm1")) {
+		new_mode = PORT_PWM1;
+
+
+#elif defined(BOARD_HAS_PWM) && BOARD_HAS_PWM >= 6
```
Comment (kd0aij) [inline src/drivers/px4fmu/fmu.cpp]: @julianoes this elif block will never be executed. Seems like the compiler should have caught such a problem.

Comment (julianoes) [inline src/drivers/px4fmu/fmu.cpp]: Yes, that was wrong, fixed with https://github.com/PX4/Firmware/pull/5719
